pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -81,6 +86,31 @@ func (r *ReconcileNMstate) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	if err = r.syncKubernetesNMStateCRDs(instance); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	reqLogger.Info("Reconcile called")
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileNMstate) syncKubernetesNMStateCRDs(instance *nmstatev1alpha1.NMstate) error {
+	var err error
+	objs := []*uns.Unstructured{}
+
+	data := render.MakeRenderData()
+	objs, err = render.RenderDir(filepath.Join("/bindata", "kubernetes-nmstate", "crds"), &data)
+	if err != nil {
+		return fmt.Errorf("failed to render kubernetes-nmstate crd: %v", err)
+	}
+
+	for _, obj := range objs {
+		if err = controllerutil.SetControllerReference(instance, obj, r.scheme); err != nil {
+			return fmt.Errorf("failed to set owner reference: %v", err)
+		}
+		if err = apply.ApplyObject(context.TODO(), r.client, obj); err != nil {
```

Comment:
As say let's investigate if we can do this using a library so we don't have to maintain apply/merge code.